code/game/objects/structures/crates_lockers/closets/tardis.dm

```diff
@@ -0,0 +1,81 @@
+var/global/obj/structure/closet/tardis/the_tardis
+var/global/obj/effect/tardis_exit/the_tardis_exit
+
+/obj/structure/closet/tardis
+	name = "T.A.R.D.I.S."
+	desc = "Bigger on the inside than on the outside."
+	icon_state = "emergency"
+	icon_closed = "emergency"
+	icon_opened = "emergencyopen"
+
+/obj/structure/closet/tardis/New()
+	..()
+	the_tardis = src
+
+/obj/structure/closet/tardis/close()
+	. = ..()
+	var/obj/effect/landmark/W
+	for(var/obj/effect/landmark/L in landmarks_list)
```

Comment:
you could just use the /area/tardis area along with one of the "safe teleport" procs to move mobs there